service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendController.java

```diff
@@ -0,0 +1,136 @@
+    private TaskDispatcher taskDispatcher;
+    private RandomIdGeneratorService randomIdGeneratorService;
+    private final TimestampService timestampService;
+    private final FhirParseService fhirParseService;
+
+    @PostMapping("/{conversationId}")
+    public ResponseEntity<String> scheduleEhrExtractResend(@PathVariable String conversationId) {
+        EhrExtractStatus ehrExtractStatus = ehrExtractStatusRepository.findByConversationId(conversationId).orElseGet(() -> null);
+
+        if (ehrExtractStatus == null) {
+            var details = getCodeableConcept(INVALID_IDENTIFIER_VALUE);
+            var diagnostics = "Provide a conversationId that exists and retry the operation";
+
+            var operationOutcome = createOperationOutcome(OperationOutcome.IssueType.VALUE,
+                                                          OperationOutcome.IssueSeverity.ERROR,
+                                                          details,
+                                                          diagnostics);
+            var errorBody = fhirParseService.encodeToJson(operationOutcome);
+
+            return new ResponseEntity<>(errorBody, HttpStatus.NOT_FOUND);
+        }
+
+        if (noErrorsInEhrReceivedAcknowledgement(ehrExtractStatus) && ehrExtractStatus.getError() == null) {
+
+            var details = getCodeableConcept(INTERNAL_SERVER_ERROR);
+            var diagnostics = "The current resend operation is still in progress. Please wait for it to complete before retrying";
+            var operationOutcome = createOperationOutcome(OperationOutcome.IssueType.BUSINESSRULE,
+                                                          OperationOutcome.IssueSeverity.ERROR,
+                                                          details,
+                                                          diagnostics);
+            var errorBody = fhirParseService.encodeToJson(operationOutcome);
+            return new ResponseEntity<>(errorBody, HttpStatus.FORBIDDEN);
+        }
+
+        LOGGER.info("Creating tasks to start the EHR Extract process resend");
+        var updatedEhrExtractStatus = prepareEhrExtractStatusForNewResend(ehrExtractStatus);
+        ehrExtractStatusRepository.save(updatedEhrExtractStatus);
+        createGetGpcStructuredTask(updatedEhrExtractStatus);
+
+        return new ResponseEntity<>(HttpStatus.ACCEPTED);
+    }
+
+    private static CodeableConcept getCodeableConcept(String codeableConceptCode) {
+        var details = new CodeableConcept();
+        var codeableConceptCoding = new Coding();
+        codeableConceptCoding.setSystem("http://fhir.nhs.net/ValueSet/gpconnect-error-or-warning-code-1");
+        codeableConceptCoding.setCode(codeableConceptCode);
+        details.setCoding(List.of(codeableConceptCoding));
+        return details;
+    }
+
+    private static boolean noErrorsInEhrReceivedAcknowledgement(EhrExtractStatus ehrExtractStatus) {
```

Comment:
boolean methods should start with an an auxillary verb like "has", "is", etc..

Reply:
thx, will rename the method